### ISSUE TITLE
[Parson] build failure: Cannot find -lparson

### ISSUE THREAD

**Issue by CIPop** [microsoft/vcpkg#32976]
### Operating system

Linux

### Compiler

GCC

### Steps to reproduce the behavior

```Shell
See instructions how to use Azure IoT SDK samples. A complete bug description is available here: https://github.com/Azure/azure-iot-sdk-c/issues/2487

The likely regression was introduced in https://github.com/microsoft/vcpkg/pull/27885
```

### Failure logs

CMake generated make files have the wrong parson path. Other libraries have the correct vcpkg path prepended:

 /mnt/e/vcpkg/installed/x64-linux/debug/lib/libz.a  -lm  __-lparson__  /mnt/e/vcpkg/installed/x64-linux/debug/lib/libfmtd.a

ld returned 1 exit status ninja: build stopped: subcommand failed.

**Comment by CIPop** [comment]
Original issue: https://github.com/Azure/azure-iot-sdk-c/issues/2487

Likely regression: https://github.com/microsoft/vcpkg/pull/27885

**Comment by vaavva** [comment]
I think I found the issue: This line should be changed
https://github.com/microsoft/vcpkg/blob/master/ports/parson/portfile.cmake#L19
from
`vcpkg_cmake_config_fixup()`
to 
`vcpkg_cmake_config_fixup(PACKAGE_NAME parson CONFIG_PATH share/parson)`

so that the generated files can be properly found

**Comment by dg0yt** [comment]
> from
> `vcpkg_cmake_config_fixup()`
> to
> `vcpkg_cmake_config_fixup(PACKAGE_NAME parson CONFIG_PATH share/parson)`

No, this would just repeat the default values (based on port name) and not make any difference.

AFAICT `-lparson` is the result of a user-side `target_link_libraries(foo PRIVATE parson)`: the word is passed as namespec to the linker.
But what this CMake config provides is an imported target `parson::parson` which would be used as `target_link_libraries(foo PRIVATE parson::parson)`. And CMake would pass the full path to the linker.

So the error probably requires looking at the user project (and it imported targets), not at port parson. Either something is not using the CMake config at all, or it is not using the imported target.
